src/assertions/OrderAssertions.ts

```diff
@@ -158,11 +156,23 @@ export class OrderAssertions {
     orders: (TakerWalletOrder | ZeroExSignedFillOrder)[],
   ) {
     let makerTokensUsed = SetProtocolUtils.CONSTANTS.ZERO;
+    const requiredComponentFills: { [addr: string]: BigNumber } = {};
 
+    const {
+      makerTokenAmount,
+      quantity,
+    } = signedIssuanceOrder;
+
+    // Validate each of the orders and calculate fill amounts
     await Promise.all(
       _.map(orders, async (order: any) => {
+        // Count up components of issuance order that have been filled from this liquidity order
+        this.accumulateComponentAmounts(order, requiredComponentFills);
```

Comment:
I don't like how this function has side effects. We can improve this by returning the object and setting it to the accumulated figure.
```
requiredComponentFills = this.accumulateComponentAmounts(order, requiredCompnentFills);
```